### PR TITLE
Add support for sending messages from NUsight

### PR DESCRIPTION
**Pull request by JosephusPaye**
This adds support for sending messages from NUsight onto the network.

Added a new method `send(message: Message, target?: string): void` on `src/client/network/network.ts`, used by components in their local network file.

Informally tested using a simulator in #208.

### CODE REVIEW

File: src/client/network/network.ts [hunk 1]
```diff
@@ -44,4 +50,16 @@ export class Network {
     }
     this.offNUClearMessages.clear()
   }
+
+  /**
+   * Send the given message on the network
+   */
+  send(message: Message, target?: string): void {
```
Comment (TrentHouliston) [inline src/client/network/network.ts]: Why make a new Message type with a target as a 2nd parameter here over using NUClearNetSend?

File: src/server/nuclearnet/fake_nuclearnet_server.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { NUClearNetSend } from 'nuclearnet.js'
 import * as XXH from 'xxhashjs'
 
 import { createSingletonFactory } from '../../shared/base/create_singleton_factory'
+import { hashType } from '../../shared/nuclearnet/nuclearnet_proxy_parser'
```
Comment (TrentHouliston) [inline src/server/nuclearnet/fake_nuclearnet_server.ts]: I don't know where this function should go, both of these places seem like a bad spot.
How do we not have a utility location yet :P

File: src/shared/nuclearnet/nuclearnet_proxy_parser.ts [hunk 3]
```diff
@@ -35,8 +36,17 @@ export class Encoder {
           case 'unlisten':
             return callback([JSON.stringify(packet)])
 
+          case 'packet': {
+            const { id, data: [key, { target, type, payload, reliable }] } = packet
+            return callback([
+              JSON.stringify({ id, nsp, key, header: { target, type, reliable } }),
```
Comment (TrentHouliston) [inline src/shared/nuclearnet/nuclearnet_proxy_parser.ts]: You're sending `type` as a string here, and adding a hashType below that isn't a part of a NUClearNetSend packet.
The decoding logic below will take that and add a .hash element to it afterwards that'll be ignored, so the hashType(type) here is doing nothing.
However, NUClearNetSend does accept a buffer or a string here, so maybe instead don't send the type in json and in the parser below add it as the type.

Comment (JosephusPaye) [reply src/shared/nuclearnet/nuclearnet_proxy_parser.ts]: Updated the decoder to not add the unused hash.

File: src/server/nuclearnet/fake_nuclearnet_server.ts (hunk 2 above)
Comment (JosephusPaye) [reply src/server/nuclearnet/fake_nuclearnet_server.ts]: Moved back :)

File: src/client/network/network.ts (hunk 1 above)
Comment (JosephusPaye) [reply src/client/network/network.ts]: Changed to accept a `NUClearNetSend`.

File: src/server/nuclearnet/fake_nuclearnet_server.ts (hunk 2 above)
Comment (BrendanAnnable) [reply src/server/nuclearnet/fake_nuclearnet_server.ts]: > How do we not have a utility location yet :P

NEVER!!! 🙂

**Pull request merged**